linode/manager
# fix: [M3-7604] - Linode Detail Summary styles and cleanup

## Description 📝
This PR cleans up the layout styles/markup for the LinodeDetail summary panel.

The initial intent was a simple tightening of the layout composition, but I also ended up extracting the header, body & footer to their own component files to improve readability.


![Screenshot 2024-01-08 at 11 21 28](https://github.com/linode/manager/assets/130582365/ee311ef1-c4b9-4037-80d8-460b1978e2d4)



## Changes  🔄
- Fix the visual issue with the "Summary" panel being misaligned with the "Public IP Addresses" & "Access" panels
- Improve responsive styling
- Improve markup
- Break down into multiple components (no change to broken files other than the ones listed above)

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-01-10 at 16 02 46](https://github.com/linode/manager/assets/130582365/0bbc4fdc-86cb-4d75-81ff-d6bbed3cc5ee) | ![Screenshot 2024-01-10 at 16 03 03](https://github.com/linode/manager/assets/130582365/34303490-eae2-456d-86b2-e16c1e44be35) |
| ![Screenshot 2024-01-10 at 16 31 23](https://github.com/linode/manager/assets/130582365/2291bb9d-a066-4c0c-a7ee-b634d85268c2) | ![Screenshot 2024-01-10 at 16 29 34](https://github.com/linode/manager/assets/130582365/5b22083f-3497-4575-968a-0f69a7c4c16d) |

## How to test 🧪

### Verification steps 
- Navigate to a linode detail and confirm layout at various breakpoints
- Confirm no regressions related to the refactor

## As an Author I have considered 🤔

*Check all that apply*
